.github/workflows/pytest.yml

```
@@ -0,0 +1,62 @@
+# Adapted from: https://github.com/marketplace/actions/install-poetry-action
+
+name: test
+
+on: pull_request
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ "3.10", "3.11", "3.12" ]
+      fail-fast: true
+    steps:
+      #----------------------------------------------
+      #       check-out repo and set-up python
+      #----------------------------------------------
+      - name: Check out repository
+        uses: actions/checkout@v4
+      - name: Set up python
+        id: setup-python
+        uses: actions/setup-python@v5
+        with:
+          python-version:
```

You need to provide this step with a python version:
```suggestion
          python-version: ${{ matrix.python-version }}
```
Otherwise the GitHub action will always run with the same fallback version of python.